File: Funciones/22. Tabla_5_for/js/tabla.js
/**
 * tabla del 5 con For
 * Autor: Santiago GM
 * Fecha 03/04/2024
 */

let tabla;
//Como paramétro

function tablas(ptablas){
    tabla = ptablas
    let contador;
    let limite = 5;
    let resultado;
    let msg = "";

    for(contador=1; limite >= contador; contador ++ ){

        resultado = tabla * contador;

        msg += `${tabla} x ${contador} = ${resultado}\n`;
    }
    return msg;
}

//Como Expresión
const tablasExp = function (ptablas){
    tabla = ptablas
    let contador;
    let limite = 5;
    let resultado;
    let msg = "";

    for(contador=1; limite >= contador; contador ++ ){

        resultado = tabla * contador;

        msg += `${tabla} x ${contador} = ${resultado}\n`;
    }
    return msg;
}

File: python/22. Tabla_5_for/utils/tabla.py
# tabla del 5 con For
# Autor: Santiago GM

#Como paramétro

def tabla_del_5(tabla_p, limite_p):
    resultado = ""
    for contador in range(1, limite_p + 1):
        resultado += f"{tabla_p} x {contador} = {tabla_p * contador}\n"
    return resultado



#función lambda
tabla_del_5_lamb = lambda tabla_p, limite_p: [f"{tabla_p} x {contador } = {tabla_p * contador}" for contador in range(1, limite_p + 1 )]


